include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2477,569 +2496,520 @@ namespace internal
-                                        global_vector_ptr
-                                          [index_array_hermite[v][2 * i] *
-                                             VectorizedArrayType::size() +
-                                           comp * static_dofs_per_component *
-                                             VectorizedArrayType::size() +
-                                           dof_info.component_dof_indices_offset
-                                               [active_fe_index]
-                                               [first_selected_component] *
-                                             VectorizedArrayType::size() +
-                                           indices[v] + v],
-                                        global_vector_ptr
-                                          [index_array_hermite[v][2 * i + 1] *
-                                             VectorizedArrayType::size() +
-                                           comp * static_dofs_per_component *
-                                             VectorizedArrayType::size() +
-                                           dof_info.component_dof_indices_offset
-                                               [active_fe_index]
-                                               [first_selected_component] *
-                                             VectorizedArrayType::size() +
-                                           indices[v] + v],
-                                        grad_weight[v]);
-                          }
+                                              2 * comp * dofs_per_face],
+                                        global_vector_ptr + ind1 +
+                                          comp * static_dofs_per_component *
+                                            VectorizedArrayType::size() +
+                                          dof_info.component_dof_indices_offset
+                                              [active_fe_index]
+                                              [first_selected_component] *
+                                            VectorizedArrayType::size(),
+                                        global_vector_ptr + ind2 +
+                                          comp * static_dofs_per_component *
+                                            VectorizedArrayType::size() +
+                                          dof_info.component_dof_indices_offset
+                                              [active_fe_index]
+                                              [first_selected_component] *
+                                            VectorizedArrayType::size(),
+                                        grad_weight,
+                                        indices,
+                                        indices);
+                        }
+                      else
+                        {
+                          Assert(false, ExcNotImplemented());
```

Comment:
I have removed the else-statments and left it only for `MatrixFreeFunctions::DoFInfo::IndexStorageVariants::contiguous`.